compiler/parser/src/test.rs

```diff
@@ -14,211 +14,161 @@
 // You should have received a copy of the GNU General Public License
 // along with the Leo library. If not, see <https://www.gnu.org/licenses/>.
 
-use crate::{ParserContext, SpannedToken, tokenizer};
+use crate::tokenizer;
 
-use leo_ast::{NodeBuilder, Statement};
-use leo_errors::{LeoError, emitter::Handler};
+use leo_ast::NodeBuilder;
+use leo_errors::{
+    LeoError,
+    emitter::{BufferEmitter, Handler},
+};
 use leo_span::{
     source_map::FileName,
-    symbol::{SessionGlobals, create_session_if_not_set_then},
-};
-use leo_test_framework::{
-    Test,
-    runner::{Namespace, ParseType, Runner},
+    span::BytePos,
+    symbol::{create_session_if_not_set_then, with_session_globals},
 };
-use serde::Serialize;
-use tokenizer::Token;
-use toml::Value;
-
-type CurrentNetwork = snarkvm::prelude::MainnetV0;
 
-// TODO: Enable parser warnings for passing tests
+use snarkvm::prelude::TestnetV0;
 
-struct TokenNamespace;
+use itertools::Itertools as _;
+use serde::Serialize;
+use serial_test::serial;
+use std::fmt::Write as _;
 
-impl Namespace for TokenNamespace {
-    fn parse_type(&self) -> ParseType {
-        ParseType::Line
-    }
+/// id and span fields just clutter up the output and
+/// may produce spurious diffs.
+fn remove_id_and_span(value: &mut serde_json::Value) {
```

Comment:
Doesn't checking the span and ID allow us to check the determinism of the parse trees?